docs/src/main/sphinx/connector/iceberg.rst

```diff
@@ -130,6 +130,48 @@ supports the following features:
 
 .. include:: alter-mv-set-properties-unsupported.fragment
 
+.. _iceberg-alter-table-execute:
+
+ALTER TABLE EXECUTE
+^^^^^^^^^^^^^^^^^^^
+
+The connector supports the following commands for use with
+:ref:`ALTER TABLE EXECUTE <alter-table-execute>`.
+
+optimize
+~~~~~~~~
+
+The ``optimize`` command is used for rewriting the active content
+of the specified table so that it is merged into potentially
+fewer larger files.
+In case that the table is partitioned, the data compaction
+acts separately on each partition selected for optimization.
+This operation improves read performance and reduces disk usage.
+
+All files with a size below the optional ``file_size_threshold``
+parameter (default value for the threshold is ``100MB``) are
+merged:
```

Comment:
I think this introduction is better than what we had before in Hive, can you apply this there as well?

Reply:
I slightly did a rewording for Hive because the optimize operation applies only for non-transactional tables.